### PR TITLE
Add a circleci config that will test ruby/rails combos

### PR DESCRIPTION
**Pull request by TheMetalCode**
Addresses #17

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,145 +1,376 @@
 version: 2.1
 
 orbs:
-  gem: doximity/gem-publisher@0
```
Comment (TheMetalCode) [inline .circleci/config.yml:4]: Removed gem publishing. See https://github.com/doximity/works-on-my-machine/tree/master for our new way to handle open source gem releases

File: .circleci/config.yml [hunk 2]
```diff
@@ -1,145 +1,376 @@
...
-            - gem-build
-          context: artifact_publishing
+              ruby-version:
+                - '3.1'
+                - '3.2'
+                - '3.3'
+              gem-version:
+                - rails-6.1.7
+                - rails-7.0.8
+                - rails-7.1.3
```
Comment (TheMetalCode) [inline .circleci/config.yml:374]: Rails 7.1 tests currently fail: https://app.circleci.com/pipelines/github/doximity/super_spreader?branch=jjh%2FIA-2150%2Fruby-rails-mvt

**Review by criswell — APPROVED** [review]

File: .circleci/config.yml (hunk 2 above)
Comment (benjaminoakes) [reply .circleci/config.yml:374]: It has something to to with `Minitest`, `assert`, etc.  It's not completely obvious to me why that is.

Comment (benjaminoakes) [reply .circleci/config.yml:374]: What do you think about removing Rails 7.1 from this matrix for now and making a new issue to follow up on the error?

Comment (TheMetalCode) [reply .circleci/config.yml]: Good idea. Filed https://github.com/doximity/super_spreader/issues/60

**Review by benjaminoakes — APPROVED** [review]

**Pull request merged**